src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -300,6 +344,23 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
     );
   }
 
+  private get isFacetSearchAvailable() {
+    if (this.areValueCaptionsSpecified) {
+      return false;
+    }
+
+    if (!this.options.enableFacetSearch) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private get areValueCaptionsSpecified() {
+    const valueCaptions = this.options.valueCaption;
+    return Object.keys(valueCaptions).length !== 0;
```

Comment:
I'd prefer if we were using underscore.js `keys` function, since it protects better against `keys(null)` or `keys(undefined)`, where the native one throws an exception.